### ISSUE TITLE
Incompatible with Microsoft.NETCore.UniversalWindowsPlatform 6.0.2

### ISSUE THREAD

**Issue by rigdern** [Reactive-Extensions/Rx.NET#452]
I created a new project and added the following two NuGet packages:
  - `Microsoft.NETCore.UniversalWindowsPlatform to 6.0.2`
  - `System.Reactive 3.1.1`

This resulted in 64 build errors. Here's a few of them:
  - `Predefined type 'System.Void' is not defined or imported`
  - `Predefined type 'System.Object' is not defined or imported`
  -
    ```
    Multiple assemblies with equivalent identity have been imported:

    'C:\Users\rigdern\.nuget\packages\Microsoft.NETCore.UniversalWindowsPlatform\6.0.2\ref\uap10.0.15138\System.Runtime.InteropServices.WindowsRuntime.dll'
      and
    'C:\Users\rigdern\.nuget\packages\System.Runtime.InteropServices.WindowsRuntime\4.0.1\ref\netcore50\System.Runtime.InteropServices.WindowsRuntime.dll'
    
    Remove one of the duplicate references.
    ```

You can find the project I used to repro this here: https://github.com/rigdern/UniversalWindowsPlatformReactiveConflict

Adam Comella
Microsoft Corp.

**Comment by onovotny** [comment]
@theesj @unnir any ideas?

**Comment by terrajobst** [comment]
Hey @jfree, who can investigate this?

**Comment by joshfree** [comment]
@joperezr or @safern could you take a look?

**Comment by joperezr** [comment]
hey @rigdern, which version of VS did you use to create the project? I just want to know which version of toolset you used to try to understand a bit more of the problem here. I'll try to clone the project and do a restore to get more info.

**Comment by rigdern** [comment]
@joperezr Here's some version information:

```
Microsoft Visual Studio Enterprise 2017 
Version 15.3.5
VisualStudio.15.Release/15.3.5+26730.16
Microsoft .NET Framework
Version 4.7.02556

Visual C# 2017   00369-60000-00001-AA381
Microsoft Visual C# 2017

NuGet Package Manager   4.3.1

Visual Studio Tools for Universal Windows Apps   15.0.26730.08
```

**Comment by joperezr** [comment]
I see, so I just downloaded your repro project and tried building and it worked just fine for me. I believe your issue is that you don't have the updated conflict resolution logic that shipped in VS, I'm using version 15.4.5 and I'm able to build just fine. Can you try updating your version of VS and trying again?
